pkg/controller/common.go

```diff
@@ -136,3 +172,349 @@ func cleanUpAzVolumeAttachments(ctx context.Context, client client.Client, azCli
 func shouldCleanUp(attachment v1alpha1.AzVolumeAttachment, mode cleanUpMode) bool {
 	return mode == all || (attachment.Spec.RequestedRole == v1alpha1.PrimaryRole && mode == primaryOnly) || (attachment.Spec.RequestedRole == v1alpha1.ReplicaRole && mode == replicaOnly)
 }
+
+func isAttached(attachment *v1alpha1.AzVolumeAttachment) bool {
+	return attachment != nil && attachment.Status.Detail != nil && attachment.Status.Detail.PublishContext != nil
+}
+
+func isCreated(volume *v1alpha1.AzVolume) bool {
+	return volume != nil && volume.Status.Detail != nil && volume.Status.Detail.ResponseObject != nil
+}
+
+func requestedToBeDeleted(azVolumeAttachment *v1alpha1.AzVolumeAttachment) bool {
```

Comment:
Consider `deletionRequested`. Also if this function took `*v1.ObjectMeta` as a parameter, it could be used with any K8s object.